androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -186,7 +186,16 @@ class AgendaDetailsViewModel(
     }
 
     private fun deleteAttendee(id: String) {
-        // TODO
+        val agendaItem = screenStateFlow.value.agendaItem
+
+        if (agendaItem !is Event) {
+            return
+        }
+
+        val newAttendees = agendaItem.attendees.toMutableList()
+        if (newAttendees.removeIf { it.userId == id }) {
+            _screenStateFlow.update { it.copy(agendaItem = agendaItem.copy(attendees = newAttendees)) }
+        }
```

Comment:
I had a similar scenario where I needed lots of such type casts. I ended up creating this function:

```
private fun updateDetailsIfEvent(update: (AgendaItemDetails.Event) -> AgendaItemDetails.Event): AgendaItemDetails {
    return when (val details = state.value.details) {
        is AgendaItemDetails.Event -> update(details)
        else -> details
    }
}
```

Works like this:

```
state = state.copy(
    details = updateDetailsIfEvent { it.copy(photos = emptyList()) }
)
```

Reply:
I agree with you here. I’ve refactored the code to something more readable.
```
_screenStateFlow.update { currentState ->
    val updatedAgendaItem = updateDetailsIfEvent { event ->
        event.copy(attendees = event.attendees.filterNot { it.userId == id })
    }
    currentState.copy(agendaItem = updatedAgendaItem)
}
```

My previous comment was only a suggestion. Early returns are actually an effective way to reduce nested if blocks.